Introduce pull request checklist (#7329)
DataDog/dd-trace-java: CONTRIBUTING.md

      to `Add New` -> `Gradle` -> `Environmental Variables`

</details>

## Pull Request Guidelines

### Title Format

Pull request titles should briefly describe the proposed changes in a way that makes sense for the users.
They should be a sentence starting with an infinitive verb, and avoid using prefixes like `[PROD]` or `PROD - ` in favor of [labels](#labels).

>[!CAUTION]
> Don't title:
> * _Another bug fix_: it doesn't describe the change 
> * _Span sampling bug fix_: it doesn't start with an infinite verb
> * _Fix off-by-one error from rule parsing_: it doesn't make sense for the user
> * _[CORE] Fix span sampling rule parsing_: it doesn't use label for component tagging
> * _Fix span sampling rule parsing when using both remote config and property config_: it doesn't fit and will be cut during changelog generation

>[!TIP]
> Do instead: _Fix span sampling rule parsing_

>[!NOTE]

> > In case of the changes can't make sense for the users, don't forget to add the [`tag: no release note` label](#labels).

Reviewer: ```suggestion
> If the changes don't make sense for the users, add the [`tag: no release note` label](#labels).
```
